hpc0.qmd

```diff
@@ -7,279 +7,1186 @@ format:
+- Create some directories and files as follows:
+```{.bash code-line-numbers=false}
+mkdir -P test0/test01
+mkdir test0/test02
+touch test0/test01/test_file
+```
+- Copy `test0/test01` into `test0/test03`
+- Move `test0/test02` into `test0/test04`
+- Delete `test0/test03`
+- Direct the output of `ls` into `ls.out`
+- Append the output of `ls -al` into `ls.out`
+- Next we will be using `cat` and `less` to look at the contents of `ls.out`...
+  - So if you are waiting then look into using these commands...
+
+
+## Check point 2 {.smaller}
+- Any questions?
+
+::: {.notes}
+- Is it time for a break?
+:::
+
+
+## Yet more commands {.smaller}
+| Command | Description |
+| :--- | :--- |
+| `cat` | Concatenate and print |
+| `less` | Open into a paginated reader view |
+| `sort` | Sort lines |
+| `cut` | Cut lines based on a given character |
+| `head` | Print the first lines of a file (default 10) |
+| `tail` | Print the last lines of a file (default 10) |
+| `wc` | Print newline, word, and byte counts for each FILE, and a total line if more than one FILE is specified. A word is a non-zero-length sequence of characters delimited by white space |
+| `grep` | Filter lines |
+| `sed` | Stream editor |
+
+::: {.notes}
+- Let's look at these in turn...
+:::
+
+
+## git clone exercise {.smaller}
+- Get a copy of some data using the following `git` command:
+```{.bash code-line-numbers=false}
+git clone https://github.com/ARCTraining/shell-training/
+```
+- git is a distributed version control system
+- Change directory into the directory called `shell-training`.
+- List the directory contents.
+
+
+## `cat`, `sort` and `cut` {.smaller}
+- The `cat` command will concatenate and print to standard output.
+  - This can be used to stream one thing after another.
+- The `sort` command will sort lines (by default alphabetically).
+- The `cut` command will cut lines based on a given character.
+- Commands that create an output stream can have the output piped into another command that can process the stream using the symbol '`|`'. 
+  - For example two files `test` and `test2` can be concatenated, sorted alphabetically and piped into head as follows: 
+```{.bash code-line-numbers=false}
+cat test test2 | sort | head
+```
+
+
+## `cat`, `sort` and `cut` demo {visibility="hidden"}
+::: {.notes}
+- Offer a demo?
+:::
+
+
+## `cat`, `sort` and `cut` exercise {.smaller}
 
 ::: {.panel-tabset}
 
 ### Problem
+- Combine `cat`, `cut`, and `sort` to print out the Latin names from the file `IOM-animals/birds.txt` in alphabetical order.
+- Save the output to a new file.
+Hint: `cut -d ',' -f 2` will split lines around the comma and yield the second field.
 
-`shell-training/data/` contains 300 data files, each of which should contain 100 values. One of these files is missing some data though. . .
-
+### Solution
+```{.bash code-line-numbers=false}
+cat birds.txt | sort | cut -d ',' -f 2 > sorted-birds.txt
+```
+\
```

Comment:
Formatting oddness with that bullet point.  Wonder if it's neater ditching the backslashes and bullet point, and just making this next "Another option" line a string without a bullet point?